snuba/query/matchers.py

```diff
@@ -323,4 +323,36 @@ def match(self, node: AnyType) -> Optional[MatchResult]:
         return result
 
 
+@dataclass(frozen=True)
+class TransformedColumn(Pattern[FunctionCallExpr]):
+    """
+    Matches a Column that has been wrapped in zero or more Functions. It recursively
+    checks the parameters of the FunctionCall to see if at least one of them is a Column
+    matching the provided column name.
+
+    Note that this makes no guarantees about the functions being called, it naively
+    goes through all the parameters and returns the first column match it finds.
+    """
+
+    column: Optional[Column] = None
```

Comment:
I don't think this class is useful without providing a Column, so it should not be optional and it should not have a default.

Also I think we should allow a match that is more complex than a column:
itt would be great if we could match
`ifNull(arrayElement(tags.values, indexOf(tags.key, 'asdasd')), '')`
as `arrayElement(tags.values, indexOf(tags.key, 'asdasd'))`. 
Though I would be ok in keeping this last part for a separate PR.

Reply:
I'll leave it to another PR for this change. Might be as simple as changing `column: Column` to be `pattern: Pattern[Expression]`.